src/packages/debugging.md

```diff
@@ -0,0 +1,59 @@
+# Debugging packages
+
+Default CheriBSD package repositories are built with optimisation levels defined
+in CheriBSD ports or upstream third-party software projects.
+Most ports use the `-O2` optimisation level but some switch to `-O3` for better
+performance or `-O1` due to compiler issues preventing them from using
+a higher optimisation level.
+Unfortunately, the FreeBSD ports build system does not allow to build ports with
+optimisations and debug symbols at the same time.
```

Comment:
I wonder if, in this paragraph, we should reduce a little bit the detail on optimisation and increase the level of detail regarding needing debug symbols to use GDB to debug package crashes?

I wonder if here we should more clearly draw the link between ports and packages, as non-FreeBSD folk may not know that ports build packages. E.g., “.. to build packages with ...”?

Reply:
> I wonder if, in this paragraph, we should reduce a little bit the detail on optimisation and increase the level of detail regarding needing debug symbols to use GDB to debug package crashes?

Yes, I think it makes sense to focus on the debugging story here.

> I wonder if here we should more clearly draw the link between ports and packages, as non-FreeBSD folk may not know that ports build packages. E.g., “.. to build packages with ...”?

I agree.

I've updated the paragraph to fix both issues. Does it look better now?